sui_core/src/authority/temporary_store.rs

```diff
@@ -118,32 +118,60 @@ impl<S> AuthorityTemporaryStore<S> {
     /// the gas object hasn't been mutated yet. Passing in `gas_object_size` so that we can also charge
     /// for the gas object mutation in advance.
     pub fn charge_gas_for_storage_changes(
-        &self,
+        &mut self,
         gas_status: &mut SuiGasStatus,
-        gas_object_size: usize,
+        gas_object: &mut Object,
```

Comment:
Could take `Object` to avoid the clone at 133 i the caller happens to have ownership?

Reply:
We cannot unfortunately. We still need to actually deduct gas from the gas_object after this function returns.